server/issue.go

```diff
@@ -539,9 +548,9 @@ func (p *Plugin) GetCreateIssueMetadataForProjects(instanceID, mattermostUserID
 		return nil, err
 	}
 
-	teamIDList := p.conf.TeamIDList
-	if len(teamIDList) > 0 {
-		injectTeamAllowedValues(metaInfo, teamIDList)
+	teamFieldKeys := injectTeamAllowedValues(metaInfo, p.conf.TeamIDList)
+	if len(teamFieldKeys) > 0 {
+		p.cacheTeamFieldKeys(instanceID, teamFieldKeys)
 	}
```

Comment:
In the cases where the JIRA instance does not have team fields, if we skip the cache, it seems it will default to `customfield_10001`. It seems that this may cause `prreProcessTeamFields` and `updateTeamValue` to try to process with a non-existint field key. 

Just double checking that is the behavior we want as I"m not that familiar with the code here